src/internals/StreamHttpReply.cpp

```diff
@@ -15,7 +25,24 @@ ArduinoHttpServer::AbstractStreamHttpReply::AbstractStreamHttpReply(Stream& stre
    m_contentType(contentType),
    m_code(code)
 {
+   headers_head = nullptr;
+   headers_tail = nullptr;
+}
 
+ArduinoHttpServer::AbstractStreamHttpReply::~AbstractStreamHttpReply()
+{
+   delete headers_head;
+}
+
+void ArduinoHttpServer::AbstractStreamHttpReply::addHeader(const char* header) {
+   Header* h = new Header(header);
```

Comment:
I can understand you want this feature, but I want to steer clear from dynamic memory allocation for this library. 
I am trying to move everything to FixString instead of using new/delete (I even rather would use smart pointers instead of new/delete (and even the C-style free)).

If you want to use a linked list I would advise not embedding it inside a class named Header, but instead make a container type class that can hold Header classes.

Reply:
I completely understand not wanting dynamic memory allocation. What do you think about using an `#ifdef` to let the user switch between this implementation and something like #19, with the default being static?